sys/kern/kern_exec.c

```diff
@@ -1208,6 +1208,9 @@ exec_new_vmspace(struct image_params *imgp, struct sysentvec *sv)
 	}
 
 #if __has_feature(capabilities)
+#ifdef __CHERI_PURE_CAPABILITY__
+	imgp->stack = (void *)cheri_setoffset(stack_addr, ssiz);
```

Comment:
Does this need an assert that stack_addr has offset 0?

Reply:
I'd prefer to just do `stack_addr + ssiz`. Then it's also safe if the offset is non-zero as it'll fault on dereference.

Reply:
Fixed